src/Endatix.Infrastructure/Integrations/Slack/SlackClient.cs

```diff
@@ -0,0 +1,68 @@
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Endatix.Core.Entities;
+using Endatix.Core.Events;
+using Endatix.Core.Infrastructure.Domain;
+using Endatix.Core.Integrations.Slack;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Endatix.Infrastructure.Integrations.Slack;
+/// <summary>
+/// A client for communications with the Slack API.
+/// </summary>
+public class SlackClient : INotificationHandler<SubmissionCompletedEvent>, ISlackClient
+{
+    private readonly ILogger<SlackClient> _logger;
+    private readonly IRepository<Form> _formRepository;
+    private readonly SlackSettings _slackSettings;
+    public SlackClient(ILogger<SlackClient> logger, IRepository<Form> formRepository, IOptions<SlackSettings> options) {
+        _formRepository = formRepository;
+        _logger = logger;
+        _slackSettings = options.Value;
+    }
+
+    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken) {
+        if(notification.Submission.IsComplete && _slackSettings.Active){
+
+            var submissionUrl = $"{_slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/')}/forms/submissions/{notification.Submission.FormId}";
```

Comment:
A good practice we follow is to not have magic strings in the code. Please use a string constant.

One option is to define a template like e.g. `private const string SubmissionUrlTemplate = "{0}/forms/submissions/{1}"` but also just defining the string literal part as constant should be fine.

Same for the the other text messages or URLs, like `message` and in `httpClient.PostAsync`.

Reply:
Done with template constants. Added TODO comment to refactor this so that it is configurable through the UI

Reply:
Yes, good suggestion.